tests/tck-build-logic/src/main/java/org/graalvm/internal/tck/utils/InteractiveTaskUtils.java

```diff
@@ -0,0 +1,91 @@
+package org.graalvm.internal.tck.utils;
+
+import java.util.Scanner;
+import java.util.function.Function;
+
+public class InteractiveTaskUtils {
+    public static <R> R askQuestion(String question, String helpMessage, Function<String, R> handleAnswer) {
+        Scanner scanner = new Scanner(System.in);
+
+        while (true) {
+            printQuestion(question);
+
+            String answer = scanner.next();
+            if (answer.equalsIgnoreCase("help")) {
+                printHelpMessage(helpMessage);
+                continue;
+            }
+
+            try {
+                return handleAnswer.apply(answer);
+            } catch (IllegalStateException ex) {
+                printErrorMessage(ex.getMessage());
+            }
+        }
+    }
+
+    public static boolean askYesNoQuestion(String question, String helpMessage, boolean defaultAnswer) {
+        Scanner scanner = new Scanner(System.in);
+
+        while (true) {
+            printQuestion(question);
+
+            String answer = scanner.nextLine();
+            if (answer.equalsIgnoreCase("help")) {
+                printHelpMessage(helpMessage);
+                continue;
+            }
+
+            if (answer.isEmpty()) {
+                return defaultAnswer;
+            }
+
+            if (answer.equalsIgnoreCase("y")) {
+                return true;
+            }
+
+            if (answer.equalsIgnoreCase("n")) {
+                return false;
+            }
+
+            printErrorMessage("Your answer must be either y (for yes) or n (for no)");
+        }
+    }
+
+    public static void closeSection() {
+        ColoredOutput.println("------------------------------------------------------------------------------------------", ColoredOutput.OUTPUT_COLOR.BLUE);
```

Comment:
Not sure it is a good idea to hardcode ColoredOutput.
A user might have a console/terminal where those codes do not work.
You should use something like
`com.oracle.svm.driver.NativeImage#hasColorSupport` to determine if using color makes sense, and if not, make ColoredOutput use behave as nops.